python/ray/tests/test_usage_stats.py

```diff
@@ -320,30 +329,103 @@ def test_usage_stats_prompt(
-        # Usage stats is not enabled for ray.init()
+        # Usage stats is not enabled for ray.init() unless it's nightly wheel.
+        m.delenv("RAY_USAGE_STATS_ENABLED", raising=False)
+        tmp_usage_stats_config_path = tmp_path / "config5.json"
+        m.setenv("RAY_USAGE_STATS_CONFIG_PATH", str(tmp_usage_stats_config_path))
+        saved_ray_version = ray.__version__
+        saved_ray_commit = ray.__commit__
+        ray.__version__ = "2.0.0"
+        ray.__commit__ = "xyzf"
         ray.init()
         ray.shutdown()
         captured = capsys.readouterr()
         assert (
-            usage_constants.USAGE_STATS_ENABLED_BY_DEFAULT_MESSAGE not in captured.out
+            usage_constants.USAGE_STATS_ENABLED_BY_DEFAULT_FOR_RAY_INIT_MESSAGE
+            not in captured.out
         )
         assert (
-            usage_constants.USAGE_STATS_ENABLED_BY_DEFAULT_MESSAGE not in captured.err
+            usage_constants.USAGE_STATS_ENABLED_FOR_RAY_INIT_MESSAGE not in captured.out
         )
+        ray.__version__ = saved_ray_version
+        ray.__commit__ = saved_ray_commit
+
+    with monkeypatch.context() as m:
+        # Usage stats is enabled for ray.init() for nightly wheel.
+        m.delenv("RAY_USAGE_STATS_ENABLED", raising=False)
+        tmp_usage_stats_config_path = tmp_path / "config6.json"
+        m.setenv("RAY_USAGE_STATS_CONFIG_PATH", str(tmp_usage_stats_config_path))
+        saved_ray_version = ray.__version__
+        saved_ray_commit = ray.__commit__
+        ray.__version__ = "2.0.0.dev0"
+        ray.__commit__ = "xyzf"
+        ray.init()
+        ray.shutdown()
+        captured = capsys.readouterr()
+        assert (
+            usage_constants.USAGE_STATS_ENABLED_BY_DEFAULT_FOR_RAY_INIT_MESSAGE
+            in captured.out
+        )
+        ray.__version__ = saved_ray_version
+        ray.__commit__ = saved_ray_commit
+
+    with monkeypatch.context() as m:
+        m.setenv("RAY_USAGE_STATS_ENABLED", "0")
+        saved_ray_version = ray.__version__
+        saved_ray_commit = ray.__commit__
+        ray.__version__ = "2.0.0.dev0"
+        ray.__commit__ = "xyzf"
+        ray.init()
+        ray.shutdown()
+        captured = capsys.readouterr()
+        assert usage_constants.USAGE_STATS_DISABLED_MESSAGE in captured.out
+        ray.__version__ = saved_ray_version
+        ray.__commit__ = saved_ray_commit
+
+    with monkeypatch.context() as m:
+        m.setenv("RAY_USAGE_STATS_ENABLED", "1")
+        saved_ray_version = ray.__version__
+        saved_ray_commit = ray.__commit__
+        ray.__version__ = "2.0.0.dev0"
+        ray.__commit__ = "xyzf"
+        ray.init()
+        ray.shutdown()
+        captured = capsys.readouterr()
+        assert usage_constants.USAGE_STATS_ENABLED_FOR_RAY_INIT_MESSAGE in captured.out
+        ray.__version__ = saved_ray_version
+        ray.__commit__ = saved_ray_commit
+
+
+def test_is_nightly_wheel():
+    saved_ray_version = ray.__version__
```

Comment:
do this in the fixture? (otherwise later tests can be screwed up if this test fails in the middle)